src/vs/workbench/parts/terminal/electron-browser/terminalPanel.ts

```diff
@@ -44,22 +42,23 @@ export class TerminalPanel extends Panel {
 		@IThemeService private themeService: IThemeService
 	) {
 		super(TERMINAL_PANEL_ID, telemetryService);
-		this.configurationHelper = new TerminalConfigHelper(platform.platform, this.configurationService, this.themeService);
 		this.toDispose = [];
 	}
 
 	public layout(dimension: Dimension): void {
-		let cols = Math.floor(this.parentDomElement.offsetWidth / TERMINAL_CHAR_WIDTH);
-		let rows = Math.floor(this.parentDomElement.offsetHeight / TERMINAL_CHAR_HEIGHT);
-		if (this.terminal) {
-			this.terminal.resize(cols, rows);
-		}
-		if (this.ptyProcess.connected) {
-			this.ptyProcess.send({
-				event: 'resize',
-				cols: cols,
-				rows: rows
-			});
+		if (this.font.charWidth && this.font.charHeight) {
```

Comment:
Prefer early exits to keep indentation low:

```
if (!this.font.charWidth || !this.font.charHeight) {
    return;
}
```

Also should this attempt to measure at this point or would that just be rework?